QuEST/src/QuEST_validation.c

```diff
@@ -94,7 +94,8 @@ typedef enum {
     E_INVALID_TROTTER_ORDER,
     E_INVALID_TROTTER_REPS,
     E_MISMATCHING_QUREG_DIAGONAL_OP_SIZE,
-    E_DIAGONAL_OP_NOT_INITIALISED
+    E_DIAGONAL_OP_NOT_INITIALISED,
+    E_NOT_A_DIAGONAL_OPERATOR
```

Comment:
should use a more descriptive enum, like `E_PAULI_SUM_NOT_DIAGONAL`. No need for superfluous grammar like "**a** diagonal operator" here